src/main/java/org/example/springbootintro/model/ShoppingCart.java

```diff
@@ -0,0 +1,41 @@
+package org.example.springbootintro.model;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.Table;
+import java.util.Set;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity
+@Getter
+@Setter
+@Table(name = "shopping_carts")
+@SQLDelete(sql = "UPDATE shopping_carts SET is_deleted = true WHERE id =?")
+@SQLRestriction(value = "is_deleted=false")
```

Comment:
I do not think we will ever delete a shopping cart, maybe if we want to delete a user, but I think that would be expected if we delete the user - the cart will be emptied and deleted as well, but that is more of a business question, so up to you